Sources/GRPCCore/Documentation.docc/Tutorials/Route-Guide/Route-Guide.tutorial

```diff
@@ -0,0 +1,499 @@
+        the asynchronous stream of points sent by the client. For each point we check if there's
+        a feature at that point and calculate the distance between that and the last point we saw.
+        After the *client* has finished sending points we populate a `Routeguide_RouteSummary` which
+        we return in the response.
+
+        @Code(name: "Sources/RouteGuideService.swift", file: "route-guide-sec04-step08-client-streaming.swift")
+      }
+
+      @Step {
+        Finally, let's look at our bidirectional streaming RPC `RouteChat`. Here we receive a stream
+        of `Routeguide_RouteNote`s and return a stream of `Routeguide_RouteNote`s. For this RPC we
+        also need to modify the state of our service to store notes sent by the client. We'll do
+        this with a helper class to store the notes.
+
+        @Code(name: "Sources/RouteGuideService.swift", file: "route-guide-sec04-step09-bidi-streaming.swift")
+      }
+
+      @Step {
+        To implement the RPC we return a `ServerResponse.Stream<Routeguide_RouteNote>`. Like in the
+        server-side streaming RPC it's initialized with a closure for writing back messages. In
+        the body of the closure we iterate the request messages and for each one call our helper
+        class to record the note and get all other notes recorded in the same location. We then
+        write each of those notes back to the client.
+
+        @Code(name: "Sources/RouteGuideService.swift", file: "route-guide-sec04-step10-bidi-streaming.swift")
+      }
+    }
+  }
+
+  @Section(title: "Creating the server") {
+    Now that we've created the service we can create a server.
+
+    @Steps {
+      @Step {
+        First we need to download a database of known features for the service. Copy
+        `route_guide_db.json` from the [gRPC Swift repository](https://github.com/grpc/grpc-swift/tree/main/Examples/v2/route-guide)
+        and place it in the `Sources` directory.
+      }
+
+      @Step {
+        Next, we need to add it as a resource to the target. Open `Package.swift`.
+
+        @Code(name: "Package.swift", file: "route-guide-sec05-step00-package.swift")
+      }
+
+      @Step {
+        Now add a `resouces` argument to copy the `route_guide_db.json` database.
+
+        @Code(name: "Package.swift", file: "route-guide-sec05-step01-package.swift")
+      }
+
+      @Step {
+        We'll also need to add a dependency on Swift Argument Parser to the package and target.
+
+        @Code(name: "Package.swift", file: "route-guide-sec05-step02-package.swift")
+      }
+
+      @Step {
+        With that done we can now create a new file in `Sources` called `RouteGuide.swift` with the
+        following code in.
+
+        `@main` indicates that the type contains the entry point to the program. In this case,
+        because `RouteGuide` conforms to `AsyncParseableCommand`, the entry point to our program
+        is the `run()` method.
```

Comment:
Should we also explain what the `@Flag` annotation does?

Reply:
Done!